tests/data/test_guitarset.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright 2022 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import apache_beam as beam
+import itertools
+import os
+import pathlib
+
+from apache_beam.testing.test_pipeline import TestPipeline
+from typing import List
+
+from basic_pitch.data.datasets.guitarset import (
+    GuitarSetToTfExample,
+    GuitarSetInvalidTracks,
+    create_input_data,
+)
+from basic_pitch.data.pipeline import WriteBatchToTfRecord
+
+RESOURCES_PATH = pathlib.Path(__file__).parent.parent / "resources"
+TRACK_ID = "00_BN1-129-Eb_comp"
+
+
+def test_guitar_set_to_tf_example(tmpdir: str):
+    input_data: List[str] = [TRACK_ID]
+    with TestPipeline() as p:
+        (
+            p
+            | "Create PCollection of track IDs" >> beam.Create([input_data])
+            | "Create tf.Example"
+            >> beam.ParDo(
+                GuitarSetToTfExample(str(RESOURCES_PATH / "data" / "guitarset"))
+            )
+            | "Write to tfrecord" >> beam.ParDo(WriteBatchToTfRecord(tmpdir))
```

Comment:
You could alternatively, not write to file and instead [use assert_that and equal_to](https://beam.apache.org/documentation/pipelines/test-your-pipeline/#an-example-test-for-a-composite-transform)